src/Microsoft.DocAsCode.MarkdownLite/Basic/BlockRules/MarkdownHtmlBlockRule.cs

```diff
@@ -3,17 +3,166 @@
     using System.Text.RegularExpressions;
 
+    using Microsoft.DocAsCode.MarkdownLite.Matchers;
+
     public class MarkdownHtmlBlockRule : IMarkdownRule
     {
+        private static readonly Matcher InlineElementNames =
+            Matcher.CaseInsensitiveString("a") |
+            Matcher.CaseInsensitiveString("em") |
+            Matcher.CaseInsensitiveString("strong") |
+            Matcher.CaseInsensitiveString("small") |
+            Matcher.CaseInsensitiveString("s") |
+            Matcher.CaseInsensitiveString("cite") |
+            Matcher.CaseInsensitiveString("q") |
+            Matcher.CaseInsensitiveString("dfn") |
+            Matcher.CaseInsensitiveString("abbr") |
+            Matcher.CaseInsensitiveString("data") |
+            Matcher.CaseInsensitiveString("time") |
+            Matcher.CaseInsensitiveString("code") |
+            Matcher.CaseInsensitiveString("var") |
+            Matcher.CaseInsensitiveString("samp") |
+            Matcher.CaseInsensitiveString("kbd") |
+            Matcher.CaseInsensitiveString("sub") |
+            Matcher.CaseInsensitiveString("sup") |
+            Matcher.CaseInsensitiveString("i") |
+            Matcher.CaseInsensitiveString("b") |
+            Matcher.CaseInsensitiveString("u") |
+            Matcher.CaseInsensitiveString("mark") |
+            Matcher.CaseInsensitiveString("ruby") |
+            Matcher.CaseInsensitiveString("rt") |
+            Matcher.CaseInsensitiveString("rp") |
+            Matcher.CaseInsensitiveString("bdi") |
+            Matcher.CaseInsensitiveString("bdo") |
+            Matcher.CaseInsensitiveString("span") |
+            Matcher.CaseInsensitiveString("br") |
+            Matcher.CaseInsensitiveString("wbr") |
+            Matcher.CaseInsensitiveString("ins") |
+            Matcher.CaseInsensitiveString("del") |
+            Matcher.CaseInsensitiveString("img");
+
+        private static readonly Matcher _HtmlMatcher =
+            Matcher.WhiteSpacesOrEmpty +
+            (
+                // html comment
+                (
+                    Matcher.String("<!--") +
+                    (
+                        Matcher.AnyCharNot('-').RepeatAtLeast(1) |
+                        (Matcher.Char('-') + Matcher.String("->").ToNegativeTest())
+                    ) + "-->"
+                ) |
+                // none-empty element
+                (
+                    Matcher.Char('<') +
+                    (
+                        (InlineElementNames + Matcher.AnyWordCharacter.ToNegativeTest()).ToNegativeTest() +
+                        // \w+
+                        Matcher.AnyWordCharacter.RepeatAtLeast(1) +
+                        // (?!:\/|[^\w\s@]*@)
+                        (
+                            Matcher.String(":/") |
+                            ((Matcher.AnyWordCharacter | Matcher.AnyCharIn(' ', '\n', '@')).ToNegativeTest() + Matcher.AnyChar).RepeatAtLeast(0) + '@'
+                        ).ToNegativeTest() +
```

Comment:
minor: need we extract the common code for (?!(?:inline\b)\w+(?!:\/|[^\w\s@]*@)\b?